### PR TITLE
Fix kibana seeding

### PR DESCRIPTION
**Pull request by ewolinetz**
@jcantrill @sosiouxme @jimmidyson PTAL.
We now correctly seed the kibana index for a user based on their OS project access.
We also remove a user and reset their default index if their previous default was removed due to access being removed on OS.

FYI: all but the latest 3 commits are throw away :)

### CODE REVIEW

File: src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+			// cull any that are in ES but not in OS (remaining in indexPatterns)
+			remove(user, indexPatterns, esClient, kibanaIndex);
+			
+			Collections.sort(common);
+			// Set default index to first index in common if we removed the default
+			String defaultIndex = getDefaultIndex(user, esClient, kibanaIndex, kibanaVersion);
+			if ( indexPatterns.contains(defaultIndex) || StringUtils.isEmpty(defaultIndex) ) {
+				for ( String project : common ) {
+					setDefaultIndex(user, project, esClient, kibanaIndex, kibanaVersion);
+					break;
```
Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java]: i dont think this is needed. means you would only process 1 project

Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java]: common used to be a set and i didn't' update the logic.. will fix

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java [hunk 2]
```diff
@@ -128,6 +129,8 @@ public void process(RestRequest request, RestChannel channel, RestFilterChain ch
 				if(updateCache(user, token, isClusterAdmin)){
 					syncAcl();
 				}
+
+				setDashboards(user, listProjectsFor(token), roles, esClient, kibanaIndex, kibanaVersion);
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java]: move this so we don't have to make a call to get projects twice; we shouldn't be updating the dashboards unless we update the cache anyways

**Comment by ewolinetz** [comment]
@jcantrill addressed nits

**Comment by jcantrill** [comment]
LGTM [merge]

**Comment by ewolinetz** [comment]
[remerge]

**Comment by ewolinetz** [comment]
rebased [merge]

**Pull request merged**